From 8108312ea6a7b22eb0a4f2b1cc3753fbe88ca963 Mon Sep 17 00:00:00 2001
From: Fabio Erculiani <lxnay@sabayon.org>
Date: Wed, 22 May 2013 13:43:47 +0200
Subject: [PATCH] dbus-userdb-util: use LOGIND_RUNNING() instead of sd_booted()

---
 dbus/dbus-userdb-util.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/dbus/dbus-userdb-util.c b/dbus/dbus-userdb-util.c
index 16bf229..b04f112 100644
--- a/dbus/dbus-userdb-util.c
+++ b/dbus/dbus-userdb-util.c
@@ -29,8 +29,9 @@
 #include <string.h>
 
 #if HAVE_SYSTEMD
-#include <systemd/sd-daemon.h>
 #include <systemd/sd-login.h>
+#include <unistd.h>
+#define LOGIND_RUNNING() (access("/run/systemd/seats/", F_OK) >= 0)
 #endif
 
 /**
@@ -55,7 +56,7 @@ _dbus_is_console_user (dbus_uid_t uid,
   dbus_bool_t result = FALSE;
 
 #ifdef HAVE_SYSTEMD
-  if (sd_booted () > 0)
+  if (LOGIND_RUNNING())
     {
       int r;
 
-- 
1.8.2.1

