diff --git a/common/gdm-common.h b/common/gdm-common.h
index 1a0c258..a379657 100644
--- a/common/gdm-common.h
+++ b/common/gdm-common.h
@@ -27,6 +27,7 @@
 #include "gdm-common-unknown-origin.h"
 
 #define GDM_CUSTOM_SESSION  "custom"
+#define LOGIND_RUNNING() (access("/run/systemd/seats/", F_OK) >= 0)
 
 G_BEGIN_DECLS
 
diff --git a/daemon/gdm-display.c b/daemon/gdm-display.c
index 2a621df..db4ae69 100644
--- a/daemon/gdm-display.c
+++ b/daemon/gdm-display.c
@@ -33,14 +33,11 @@
 #include <glib/gi18n.h>
 #include <glib-object.h>
 
-#ifdef WITH_SYSTEMD
-#include <systemd/sd-daemon.h>
-#endif
-
 #include "gdm-display.h"
 #include "gdm-display-glue.h"
 #include "gdm-display-access-file.h"
 
+#include "gdm-common.h"
 #include "gdm-settings-direct.h"
 #include "gdm-settings-keys.h"
 
@@ -365,7 +362,7 @@ gdm_display_real_get_timed_login_details (GdmDisplay *display,
          * can support auto/timed login on auxilliary seats in the
          * systemd path.
          */
-        if (sd_booted() > 0) {
+        if (LOGIND_RUNNING()) {
                 if (g_strcmp0 (display->priv->seat_id, "seat0") != 0) {
                         goto out;
                 }
-- 
1.8.1.5

