From cbee01a7f960a0ea76ad8ee3b1377f67ba44ac92 Mon Sep 17 00:00:00 2001
From: Fabio Erculiani <lxnay@sabayon.org>
Date: Thu, 2 May 2013 14:16:50 +0200
Subject: [PATCH] gen_compile: fix iscsi module path install, s/RELEASE/KV

---
 gen_compile.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/gen_compile.sh b/gen_compile.sh
index ccc8b8d..aab2716 100755
--- a/gen_compile.sh
+++ b/gen_compile.sh
@@ -669,12 +669,12 @@ compile_iscsi() {
 		MAKE=${UTILS_MAKE} compile_generic "user" ""
 
 		# if kernel modules exist, copy them to initramfs, otherwise it will be compiled into the kernel
-		mkdir -p "${TEMP}/initramfs-iscsi-temp/lib/modules/${RELEASE}/kernel/drivers/scsi/"
+		mkdir -p "${TEMP}/initramfs-iscsi-temp/lib/modules/${KV}/kernel/drivers/scsi/"
 		for modname in iscsi_tcp libiscsi scsi_transport_iscsi
 		do
 			if [ -e "${CMD_KERNEL_DIR}/drivers/scsi/${modname}.ko" ]
 			then
-				cp ${CMD_KERNEL_DIR}/drivers/scsi/${modname}.ko "${TEMP}/initramfs-iscsi-temp/lib/modules/${RELEASE}/kernel/drivers/scsi/"
+				cp ${CMD_KERNEL_DIR}/drivers/scsi/${modname}.ko "${TEMP}/initramfs-iscsi-temp/lib/modules/${KV}/kernel/drivers/scsi/"
 			fi
 		done
 
-- 
1.8.1.5

